### ISSUE TITLE
Full KMS - #Kivy with Mesa

### ISSUE THREAD

**Issue by kuzeyron** [anholt/mesa#107]
The steps are easy. 
```
sudo apt-get install libsdl2-dev libsdl2-image-dev libsdl2-mixer-dev libsdl2-ttf-dev \
   pkg-config libgl1-mesa-dev libgles2-mesa-dev \
   python-setuptools libgstreamer1.0-dev git-core \
   gstreamer1.0-plugins-{bad,base,good,ugly} \
   gstreamer1.0-{omx,alsa} python-dev libmtdev-dev \
   xclip xsel -y

sudo pip2 install Cython==0.25.2 && \
sudo pip2 install git+https://github.com/kivy/kivy.git@master
```
And then running @frankgould's [code](https://gist.github.com/frankgould/4e217f77b4b16d147e1297e478d652f2)
I have already once edited the code to be running better. I don't think it matters what code you are using. As long as there are Fade in/out or Animations it should give you the same errors.

But if you try it. With the default mesa version that is set it will talk about Memory allocation errors and so on. FBO shows more often up if it's the Default drivers. The Memory allocation part show up in the dmesg  and it freezes the entire Desktop on the Pi(3B+).

Running mesa upgrade through "git clone" makes these errors to stop showing up.
It can run it for hours without a error. But then going back to the original mesa version. It crashes.

Regards, Mathias

**Comment by anholt** [comment]
If the current VC4 drivers in Mesa master work, then there's not much I can do for you.  Raspbian has ancient software, and I have no control over their update schedule.

**Comment by kuzeyron** [comment]
Ahaa okay! Thank you anyway for your time!

**Comment by frankgould** [comment]
Re: VC4 and Mesa Drivers Update Plan
Tue Jun 26, 2018 1:39 pm
Link:  https://www.raspberrypi.org/forums/viewtopic.php?f=66&t=216804&p=1333284#p1333284

> FrankGould wrote: ↑
> Tue Jun 26, 2018 1:07 pm
> I have been working with a team of Kivy and Mesa driver engineers who have identified that Raspbian graphics need to be updated and a new version tested and released. Below is more information about this issue that was first reported on Kivy Github Oct 7, 2016, link below.
> https://github.com/kivy/kivy/issues/4662
> 
> Below is the Final Word from one of the engineers:
> What is needed to fix this would be a updated VC4 (VideoCore IV) and then update the graphic drivers. I saw a VC4 version here on Github but latest changes are about two years old.
> 
> This is a dead end unless the foundation themself (Raspberry Pi) marks this as important. When I've been testing, I've tried using Gl4es and Mesa.
> Kivy only in Fullscreen or openGL segmentation fault:
>https://www.raspberrypi.org/forums/viewtopic.php?p=1173191
> 
> Kivy only in Fullscreen or openGL segmentation fault:
> https://www.raspberrypi.org/forums/viewtopic.php?t=185675 
> 
> Below are the issues we've reported and the support responses:
> 
> Kivy apps crash - need mesa version to fix #106:
> https://github.com/anholt/mesa/issues/106 
> 
> Full KMS - #Kivy with Mesa #107:
> https://github.com/anholt/mesa/issues/107 
> 
> Because of this Kivy driver problem, I tried to migrate my app to pi3d but here's what I reported in May 2018 and have not had any responses:
> 
> pi3d and streaming video crash:
> https://www.raspberrypi.org/forums/viewtopic.php?f=28&t=214284&p=1319167 
> 
> Where should we go to request Raspbian be scheduled for graphic drivers updates, tests, and release?

**fruitoftheloom wrote:**
http://www.anholt.net

http://www.anholt.net/resume.html

https://github.com/anholt

https://anholt.github.io/twivc4/
